receiver/receivercreator/config_expansion.go

```diff
@@ -34,16 +34,22 @@ func evalBackticksInConfigValue(configValue string, env observer.EndpointEnv) (a
 	// results if possible.
 	var expansions []any
 
-	// Loop through configValue one rune at a time using exprStartIndex to keep track of
+	// Loop through configValue one run at a time using exprStartIndex to keep track of
 	// inside or outside of expressions.
 	for i := 0; i < len(configValue); i++ {
 		switch configValue[i] {
 		case '\\':
 			if i+1 == len(configValue) {
 				return nil, errors.New(`encountered escape (\) without value at end of expression`)
 			}
-			output.WriteByte(configValue[i+1])
-			i++
+			if string(configValue[i+1]) != "`" {
```

Comment:
Not sure why we need to cast to string here
```suggestion
			if configValue[i+1] != '`' {
```